source/Calamari.Shared/Integration/Packages/NuGet/NuGetV3LibDownloader.cs

```diff
@@ -25,31 +25,27 @@ public static void DownloadPackage(string packageId, IVersion version, Uri feedU
             if (feedCredentials != null)
             {
                 var cred = feedCredentials.GetCredential(feedUri, "basic");
-                sourceRepository.PackageSource.Credentials = new PackageSourceCredential("octopus", cred.UserName, cred.Password, true);
+                sourceRepository.PackageSource.Credentials = new PackageSourceCredential("octopus", cred.UserName, cred.Password, true, null);
             }
 
             using (var sourceCacheContext = new SourceCacheContext() { NoCache = true })
             {
-                var providers = new SourceRepositoryDependencyProvider(sourceRepository, logger, sourceCacheContext);
-                var libraryIdentity = new LibraryIdentity(packageId, version.ToNuGetVersion(), LibraryType.Package);
-
+                var providers = new SourceRepositoryDependencyProvider(sourceRepository, logger, sourceCacheContext, sourceCacheContext.IgnoreFailedSources, false);
                 var targetPath = Directory.GetParent(targetFilePath).FullName;
                 if (!Directory.Exists(targetPath))
                 {
                     Directory.CreateDirectory(targetPath);
                 }
 
                 string targetTempNupkg = Path.Combine(targetPath, Path.GetRandomFileName());
-                using (var nupkgStream = new FileStream(targetTempNupkg,
-                                                        FileMode.Create,
-                                                        FileAccess.ReadWrite,
-                                                        FileShare.ReadWrite | FileShare.Delete,
-                                                        4096,
-                                                        true))
+                var packageDownloader =  providers.GetPackageDownloaderAsync(new PackageIdentity(packageId, version.ToNuGetVersion()), sourceCacheContext, logger, CancellationToken.None)
+                                                      .GetAwaiter()
+                                                      .GetResult();
+                var fileCopied = packageDownloader.CopyNupkgFileToAsync(targetTempNupkg, CancellationToken.None).GetAwaiter().GetResult();
+                
+                if (!fileCopied) //I would expect any actual standard exception to be thrown above and not returned as a bool
                 {
-                    providers.CopyToAsync(libraryIdentity, nupkgStream, CancellationToken.None)
```

Comment:
This `CopyToAsync` method appears to have been replaced by the `CopyNupkgFileToAsync` on the right